### PR TITLE
Allow disabled job to load if it contains invalid state

### PR DESCRIPTION
**Pull request by Andyz26**
### Context

Allow job cluster loading on disabled job cluster definitions if the cluster definition misses required parameters to maintain proper back compat.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobClusterDefinitionImpl.java [hunk 1]
```diff
@@ -75,14 +76,27 @@ public JobClusterDefinitionImpl(@JsonProperty("name") String name,
...
+            Preconditions.checkNotNull(labels, "labels cannot be empty.");
+            Preconditions.checkArgument(
+                labels.stream()
+                    .anyMatch(l ->
+                        l.getName().equalsIgnoreCase(SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label)),
+                "Missing required label: " + SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label);
+        }
     }
 
+    public JobClusterDefinitionImpl(String name,
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobClusterDefinitionImpl.java:89]: nit: @VisibleForTesting? Also, does this need to public or is package-private good enough?

Comment (sundargates) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobClusterDefinitionImpl.java:89]: Actually it looks like this is only called from the builder. so, maybe private is good enough.

**Review by sundargates — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobClusterDefinitionImpl.java (hunk 1 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobClusterDefinitionImpl.java:89]: it's also used in other path where the create request etc. doesn't have include data about isDisabled thus I gave it a default to maintain the current behavior.

**Review by fdc-ntflx — APPROVED** [review]
lgtm!

**Pull request merged**